src/schedule/zephyr_domain.c

```diff
@@ -52,17 +52,46 @@ struct zephyr_domain {
 	struct ll_schedule_domain *ll_domain;
 };
 
+/* perf measurement windows size 2^x */
+#define CYCLES_WINDOW_SIZE	10
+
 static void zephyr_domain_thread_fn(void *p1, void *p2, void *p3)
 {
 	struct zephyr_domain *zephyr_domain = p1;
 	int core = cpu_get_id();
 	struct zephyr_domain_thread *dt = zephyr_domain->domain_thread + core;
+	unsigned int handler_runs = 0, overruns = 0, cycle_sum = 0;
+	uint32_t cycles0, cycles1, diff, cycles_max, timer_fired;
 
 	for (;;) {
 		/* immediately go to sleep, waiting to be woken up by the timer */
 		k_sem_take(&dt->sem, K_FOREVER);
 
+		cycles0 = k_cycle_get_32();
 		dt->handler(dt->arg);
+		cycles1 = k_cycle_get_32();
+
+		if (cycles1 > cycles0)
```

Comment:
I wonder if we should do these calculations on our own - wouldn't it be better to give zephyr a callback method to report errors like budget violation?